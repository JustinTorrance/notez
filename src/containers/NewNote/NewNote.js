import React, {Component} from 'react';
import Input from '../Input/Input';
import { connect } from 'react-redux';
import { addNote } from '../../thunks/addNote';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

export class NewNote extends Component {
  constructor() {
    super()
    this.state = {
      inputs: [],
      title: '',
      listItems: []
    }
  }

  createNewItem = (text) => {
    const newListItem = { id: Date.now(), text, completed: false }
    const newInput =  <Input key={Date.now()} createNewItem={this.createNewItem} />
    const listItems = [...this.state.listItems, newListItem]
    const inputs = [...this.state.inputs, newInput]
    this.setState({ listItems, inputs })
  }
  
  handleTitleChange = (e) => {
    this.setState({ title: e.target.value })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const url = 'http://localhost:3001/api/v1/notes'
    const { title, listItems } = this.state
    this.props.addNote(url, {title, listItems})
  }

  render() {
    const mappedState = this.state.inputs.map(form => {
      return form
    })
    return(
      <form onSubmit={() =>{}} className='new-note-wrapper'>
        <input 
          className='new-note-inputs'
          type='text'
          placeholder='title'
          value={this.state.title}
          onChange={this.handleTitleChange}
        />
        <Input createNewItem={this.createNewItem} />
        {mappedState}
        <div className='new-note-button-wrapper-2'>
          <button className='new-note-submit-button' type='submit' onClick={this.handleSubmit}>Submit</button>
          <NavLink className='route-to-home-link' to='/'>
            <button className='route-to-home-button'>Return to your notes</button>
          </NavLink>
        </div>
      </form>
    )
  }
}

NewNote.propTypes = {
  addNote: PropTypes.func.isRequired
}

export const mapDispatchToProps = (dispatch) => ({
  addNote: (url, newNote) => dispatch(addNote(url, newNote))
})

export default connect(null, mapDispatchToProps)(NewNote);
