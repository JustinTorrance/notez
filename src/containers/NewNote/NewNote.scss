.new-note-wrapper {
  border: 1px solid #d3d3d3;
  min-width: 200px;
  margin: 1.5rem;
  border-radius: 4px;
  box-shadow: 1px 2px 6px grey;
  padding: 1rem;
}

.new-note-inputs {
  width: 99%;
}

.new-note-button-wrapper-2 {
  display: flex;
  justify-content: space-between;
  margin: 1rem;
}

.route-to-home-button,
.new-note-submit-button {
  border-radius: 4px;
  &:hover {
    background-color: hotpink;
    cursor: pointer;
  }
}