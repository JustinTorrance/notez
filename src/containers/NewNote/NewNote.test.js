import { NewNote, mapDispatchToProps } from './NewNote';
jest.mock('../../thunks/addNote');

describe('NewNote', () => {
  const addNote = jest.fn()
    
  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const mockNote = {title: 'Walk dog', listItems: [{id: 1, text: 'Find leash'}]};
      const mockUrl = 'www.url.com';
      const actionToDispatch = addNote(mockUrl, mockNote);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addNote(mockUrl,mockNote);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});