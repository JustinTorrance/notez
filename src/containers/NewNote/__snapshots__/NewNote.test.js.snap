// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewNote should match the snapshot 1`] = `
"<form onSubmit={[Function: onSubmit]} className=\\"new-note-wrapper\\">
  <input className=\\"new-note-inputs\\" type=\\"text\\" placeholder=\\"title\\" value=\\"\\" onChange={[Function]} />
  <Input createNewItem={[Function]} />
  <p className=\\"enter-user-messaging\\">
    Press enter for a new line
  </p>
  <div className=\\"new-note-button-wrapper-2\\">
    <button className=\\"new-note-submit-button\\" type=\\"submit\\" onClick={[Function]}>
      Submit
    </button>
    <NavLink className=\\"route-to-home-link\\" to=\\"/\\">
      <button className=\\"route-to-home-button\\">
        Return to your notes
      </button>
    </NavLink>
  </div>
</form>"
`;
